server/src/test/java/org/elasticsearch/index/codec/zstd/StoredFieldCodecDuelTests.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.codec.zstd;
+
+import org.apache.lucene.codecs.Codec;
+import org.apache.lucene.codecs.lucene99.Lucene99Codec;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.tests.index.ForceMergePolicy;
+import org.apache.lucene.tests.index.RandomIndexWriter;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.index.codec.LegacyPerFieldMapperCodec;
+import org.elasticsearch.index.codec.PerFieldMapperCodec;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class StoredFieldCodecDuelTests extends ESTestCase {
+
+    private static final String FIELD_1 = "string_field_1";
+    private static final String FIELD_2 = "binary_field_2";
+    private static final String FIELD_3 = "int_field_3";
+    private static final String FIELD_4 = "long_field_4";
+    private static final String FIELD_5 = "float_field_5";
+    private static final String FIELD_6 = "double_field_5";
```

Comment:
Nit: I'd rename these to STRING_FIELD = "string_field", BINARY_FIELD = "binary_field" etc, since we only have one of each type and we process each field in a specific way, based on their type.